src/Spectre.Angular2Client/src/app/heatmap/heatmap.component.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * heatmap.component.ts
+ * Component of single heatmap.
+ *
+ Copyright 2017 Grzegorz Mrukwa, Sebastian Pustelnik, Daniel Babiak
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import {Component, Input, OnInit} from '@angular/core';
+import {HeatmapService} from './heatmap.service';
+import {Heatmap} from './heatmap';
+
+
+
+@Component({
+  selector: 'app-heatmap',
+  templateUrl: './heatmap.component.html',
+  styleUrls: ['./heatmap.component.css']
+})
+export class HeatmapComponent implements OnInit {
+
+  @Input() public id: number;
```

Comment:
My idea was to keep this component to be a reusable general-purpose heatmap, while the data would be set and downloaded from the `preparation` component. This was supposed to be a wrapper for plotly heatmap **only**, without any additional logic behind at all.

To make long story short:

* defaults are set in this component - they belong here and allow to create this component seamlessly from the outside
* it would be good to make some properties set from the outside in the long run, e.g. width or height
* download & connection with REST API should be realized by `prepraration` component, because it's its responsibility in fact
* by default, heatmap should have 1:1 aspect ratio (pretty useful). Right now it's just filling whole canvas with known data.